pkg/postage/listener/listener.go

```diff
@@ -61,12 +61,11 @@ type BlockHeightContractFilterer interface {
 }
 
 type listener struct {
-	logger    log.Logger
-	ev        BlockHeightContractFilterer
-	blockTime time.Duration
-
+	logger              log.Logger
+	ev                  BlockHeightContractFilterer
+	blockTime           time.Duration
 	postageStampAddress common.Address
-	quit                chan struct{}
+	ctxCancel           context.CancelFunc
```

Comment:
I dont see why we need to change this. We have this type of quit handling at a bunch of places and I prefer this as it is simple and easier to understand than the context approach. I dont think this needs to be part of this change at all, so I would revert this change back.

This PR should only enable leak check and not need to change any logic around the codebase. Also if you are going through other packages I would recommend you to only focus on enabling the leak check.

Reply:
Initially when this PR was published, there was a problem because code needed to use context and quit channel to signal termination, but in code only quit channel was being used. 

First reason for that change was to use only one approach to signal termination, because it would be sufficient enough. Having to listen on two signals is error prone as users can mistakenly avoid listening on one (as manifested in this code)

Since then there was newer PR which fixed this #3671.

---

Another reasoning why quit channel was replaced with context was the same as described [here](https://github.com/ethersphere/bee/pull/3531#discussion_r1065714484).